### PR TITLE
Health check for Kafka and postgre

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Description

- Moved kafka consumer implementation to integrations project
- Added kafka connectivity check to health check 
- Added postgre health check

## Related Issue(s)
- #160

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] All tests run green

### CODE REVIEW

File: src/Altinn.Notifications.Core/Extensions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -19,7 +17,7 @@ public static class ServiceCollectionExtensions
     /// </summary>
     /// <param name="services">service collection.</param>
     /// <param name="config">the configuration collection</param>
-    public static IServiceCollection AddCoreServices(this IServiceCollection services, IConfiguration config)
+    public static void AddCoreServices(this IServiceCollection services, IConfiguration config)
```
Comment (SandGrainOne) [inline src/Altinn.Notifications.Core/Extensions/ServiceCollectionExtensions.cs:20]: why change this to void?

File: src/Altinn.Notifications.Core/Services/Interfaces/IGetOrderService.cs [hunk 2]
```diff
@@ -20,5 +20,5 @@ public interface IGetOrderService
     /// </summary>
     /// <param name="senderRef">The senders reference</param>
     /// <param name="creator">The creator of the orders</param>
-    public Task<(List<NotificationOrder> Orders, ServiceError? Error)> GetOrdersBySendersReference(string senderRef, string creator);
+    public Task<(List<NotificationOrder>? Orders, ServiceError? Error)> GetOrdersBySendersReference(string senderRef, string creator);
```
Comment (SandGrainOne) [inline src/Altinn.Notifications.Core/Services/Interfaces/IGetOrderService.cs:23]: When working with collections we should try to always have an instance, but allow for it to be empty.

File: src/Altinn.Notifications.Core/Extensions/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (acn-sbuad) [reply src/Altinn.Notifications.Core/Extensions/ServiceCollectionExtensions.cs:20]: To have the same notation for adding health check and service. 
Don't see an explicit need to the serviceCollection.

File: src/Altinn.Notifications.Integrations/Health/KafkaHealthCheck.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
+﻿using Confluent.Kafka;
+
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Altinn.Notifications.Integrations.Health;
+
+/// <summary>
+/// Health check service confirming Kafka connenctivity
+/// </summary>
+public class KafkaHealthCheck : IHealthCheck
```
[the remark this answers is not in the document]
Comment (acn-sbuad) [reply src/Altinn.Notifications.Integrations/Health/KafkaHealthCheck.cs]: made it disposeable

File: src/Altinn.Notifications.Core/Extensions/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (SandGrainOne) [reply src/Altinn.Notifications.Core/Extensions/ServiceCollectionExtensions.cs:20]: I'm not always too fond of dot chaining, but it's still common to support it.

**Review by SandGrainOne — APPROVED** [review]

**Pull request merged**